datafusion/datasource/Cargo.toml

```diff
@@ -36,6 +36,7 @@ default = ["compression"]
 
 [dependencies]
 arrow = { workspace = true }
+arrow-schema = { workspace = true }
```

Comment:
Since we already have arrow as a dependency, the arrow-schema dependency seems unecessary